crates/scroll/node/src/node.rs

```diff
@@ -0,0 +1,343 @@
+//! Node specific implementations for Scroll.
```

Comment:
I think it would be cleaner if we break this file up a bit and introduce some modules for each of the different components, i.e. pool, engine, pool etc.